src/main/java/redis/clients/jedis/resps/ClusterShardInfo.java

```diff
@@ -0,0 +1,44 @@
+package redis.clients.jedis.resps;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class holds information about a shard of the cluster with command {@code CLUSTER SHARDS}.
+ * They can be accessed via getters. There is also {@link ClusterShardInfo#getClusterShardInfo()}
+ * method that returns a generic {@link Map} in case more info are returned from the server.
+ */
+public class ClusterShardInfo {
+
+  public static final String SLOTS = "slots";
+  public static final String NODES = "nodes";
+
+  private final List<Long> slots;
+  private final List<ClusterShardNodeInfo> nodes;
+
+  private final Map<String, Object> clusterShardInfo;
+
+  /**
+   * @param map contains key-value pairs with cluster shard info
+   */
+  @SuppressWarnings("unchecked")
+  public ClusterShardInfo(Map<String, Object> map) {
+    slots = (List<Long>) map.get(SLOTS);
+    nodes = (List<ClusterShardNodeInfo>) map.get(NODES);
+
+    clusterShardInfo = map;
+  }
+
+  public List<Long> getSlots() {
```

Comment:
[DISCUSSION]

Can we do better than this for `slots`? I.e.

- `List<KeyValue<Long, Long>>` - KeyValue doesn't make sense but we'd have endpoints together
- `List<List<Long>>` - inner List will always be of size 2, containing two endpoints

Reply:
In both cases it is not obvious what the key and value or the first and second element of the internal list mean.

Maybe List<SlotRange> where SlotRange includes two fields: the range start and the range end.

Reply:
> Maybe `List<SlotRange>` where `SlotRange` includes two fields: the range start and the range end.

I would prefer avoiding another new class.

Reply:
Ok. For me the second option looks better (`List<List<Long>>`)

Reply:
vote for `List<List<Long>>`

Reply:
Done. But if you know how to make it better, let me know